core/services/keystore/solana.go

```diff
@@ -23,6 +23,30 @@ type Solana interface {
 	Sign(ctx context.Context, id string, msg []byte) (signature []byte, err error)
 }
 
+// SolanaSigner adapts Solana to [loop.Keystore].
+type SolanaSigner struct {
+	Solana
+}
+
+func (s *SolanaSigner) Keys(ctx context.Context) (accounts []string, err error) {
+	ks, err := s.GetAll()
+	if err != nil {
+		return nil, err
+	}
+	for _, k := range ks {
+		accounts = append(accounts, k.PublicKeyStr())
+	}
+	return
+}
+
+func (s *SolanaSigner) Sign(_ context.Context, id string, msg []byte) ([]byte, error) {
+	k, err := s.Get(id)
+	if err != nil {
+		return nil, err
+	}
+	return k.Sign(msg)
+}
```

Comment:
isn't this just duplicate logic of the underlying sign method in the wrapped keystore?

Reply:
Indeed. No longer necessary :partying_face: